src/kernel/include/cap_ops.h

```diff
@@ -123,8 +127,12 @@ cap_cons(struct captbl *t, capid_t capto, capid_t capsub, capid_t expandid)
 	} else {
 		/* FIXME: we need to ensure TLB quiescence for pgtbl cons/decons! */
 		u32_t flags = 0, old_pte, new_pte, old_v, refcnt_flags;
+		u32_t old_flags;
+		u32_t size_order;
 
-		intern = pgtbl_lkup_lvl(((struct cap_pgtbl *)ct)->pgtbl, expandid, &flags, ct->lvl, depth);
+		/* PRY:look up the position to add this to */
+		intern=pgtbl_lkup_pte(&(((struct cap_pgtbl *)ct)->pgtbl),expandid,&size_order,&old_flags);
```

Comment:
Depth is not included here. For cortex-m, there are two kinds of page table nodes, on kind is the top-level node, and another kind is other nodes. Thus, the level can be seen as designated when the page table nodes are created.